enhancements/insights/insights-runtime-extractor.md

```diff
@@ -0,0 +1,643 @@
+
+##### Thin Privileged Coordinator Process
+
+The risk of a workable exploit can be greatly reduced by minimizing the amount of total potential code, including the number and size of third-party dependencies. This comes from two key factors: First, the reduced complexity of the system limits the number of potential attack points, and secondly, the smaller total code set improves the speed and effectiveness of both human and automated code auditing.
+
+As such, three key key rules should be followed in the design and implementation of the coordinator process
+
+1. __Dynamic loading of arbitrary executable code should never be supported or allowed by the coordinator process.__ This precludes scripting language runtimes (such as python and JS) as well as bytecode interpreting languages such as Java, .Net, and WASM. It also precludes usage of dlopen(), or libraries with dynamic loading capabilities: the process must be statically compiled. 
+1. __Dependencies in the coordinator should be kept to only the provided language SDK, and Linux system contracts__, even at the expense of some code duplication
+1. __Coordinator code should be limited to what is strictly required__ to execute in a privileged manner. All non-privilege requiring code should be externalized into the logic of other processes.
+
+##### Defensive Input and Memory Guarding
+
+While keeping the bulk of analysis logic out of the coordinator process reduces risk, there is still the potential for a deprivileged insights-runtime-extractor component to influence the coordinator as a corrupted relay through its required communication channel. As an example, a corrupted container could lead a insights-runtime-extractor process to write a carefully crafted payload which triggers a buffer overflow and potentially arbitrary code execution in the coordinator. Therefore the coordinator process should take particular care in the handling of all input, assuming potentially hostile data. Additionally, the coordinator process should take advantage of technologies and techniques that mitigate memory corruption defects. As an example, Rust’s memory access model provides strong guarantees of memory safety, and limits state mutability to the smaller set of intended write code paths. Usage of a defensive language like Rust over a traditional memory unsafe language like C would mitigate risk further by providing a second line of defense, as well as preventing a number of common issues frequently leveraged in exploits. 
```

Comment:
You can update this to reflect on the idea of the coordinator to not read the output of the fingerprint executables.